File: lecture_007(Strings)/l_008_removeduplicates.cpp
#include<iostream>
#include<string>
using namespace std;

void removeduplicates(string s) {
    for(int i=0; i<s.size()-1; i++) {
        char r1 = s[i];
        char r2 = s[i+1];
        if(r1 != r2) {
            cout<<r1;
        }
    }
    cout<<s[s.size()-1];
}

int main() {
    string s;
    cin>>s;
    removeduplicates(s);

    return 0;
}

File: lecture_007(Strings)/l_008_removeduplicates.java
import java.util.Scanner;
public class l_008_removeduplicates {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String s = scn.nextLine();
        removeDuplicates(s);
    }
    public static void removeDuplicates(String s) {
        for(int i=0; i<s.length()-1; i++) {
            char r1 = s.charAt(i);
            char r2 = s.charAt(i+1);
            if(r1 != r2) {
                System.out.print(r1);
            }
        }
        System.out.print(s.charAt(s.length()-1));
    }
}